### PR TITLE
SelectPanel2: 3 tiny bug fixes

### PR DESCRIPTION
**Pull request by siddharthkp**
### Fix 1

In #3758, I forgot that we would need to add Suspense support to `useSlots` first before making footer optional. Default header is fine because it can't be wrapped in suspense, but footer totally can be.

Haven't tried to update useSlots yet, will explore later.

| Before | After |
|--------|--------|
| Select panel with two footers! | Select panel with one footer | 

### Fix 2 

The menu was not closing on it's own because I made a silly mistake. We will be able to catch these silly things once I write some tests 😅 

```diff
- if (props.open === 'undefined') setInternalOpen(false)
+ if (props.open === undefined) setInternalOpen(false)
```

### Fix 3

Added a warning to the ActionMenu story because it's implementation will most definitely change once we implement SelectPanel as a modal

SelectPanel story with a banner that says This implementation will most likely change. [See decision log for more details.](https://github.com/github/primer/discussions/2614#discussioncomment-6879407)

### CODE REVIEW

**Comment by joshblack** [comment]
> we would need to add Suspense support to useSlots first before making footer optional

Just wanted to ask, is it possible for `useSlots` to be Suspense-compatible? I would have thought that `React.Children` and `Suspense` would be incompatible and was curious how this will work 👀

File: src/drafts/SelectPanel2/SelectPanel.tsx [hunk 1]
```diff
@@ -55,13 +55,13 @@ const SelectPanel = props => {
...
   const onInternalClose = () => {
-    if (props.open === 'undefined') setInternalOpen(false)
+    if (props.open === undefined) setInternalOpen(false)
     if (typeof props.onCancel === 'function') props.onCancel()
   }
   // @ts-ignore todo
   const onInternalSubmit = event => {
     event.preventDefault()
-    if (props.open === 'undefined') setInternalOpen(false)
+    if (props.open === undefined) setInternalOpen(false)
```
Comment (broccolinisoup) [inline src/drafts/SelectPanel2/SelectPanel.tsx:64]: I think you fixed it in the other PR, no? https://github.com/primer/react/pull/3770

File: src/drafts/SelectPanel2/SelectPanel.tsx [hunk 2]
```diff
@@ -105,7 +105,7 @@ const SelectPanel = props => {
             {slots.header || <SelectPanel.Header />}
             {childrenInBody}
             {/* render default footer as fallback */}
-            {slots.footer || <SelectPanel.Footer />}
+            {slots.footer}
```
Comment (broccolinisoup) [inline src/drafts/SelectPanel2/SelectPanel.tsx:108]: >I forgot that we would need to add Suspense support to useSlots first before making footer optional. 

This is probably my ignorance but what is the reason of wrapping the footer in Suspense? 🤔

File: src/drafts/SelectPanel2/SelectPanel.tsx (hunk 1 above)
Comment (siddharthkp) [reply src/drafts/SelectPanel2/SelectPanel.tsx:64]: This is that PR :)

File: src/drafts/SelectPanel2/SelectPanel.tsx (hunk 2 above)
Comment (siddharthkp) [reply src/drafts/SelectPanel2/SelectPanel.tsx:108]: If we need to fetch data before rendering the list, we'd like the footer to also be hidden until then:


https://github.com/primer/react/assets/1863771/8ce5704a-8162-439a-a105-4ee4c55ce89e

File: src/drafts/SelectPanel2/SelectPanel.tsx (hunk 1 above)
Comment (broccolinisoup) [reply src/drafts/SelectPanel2/SelectPanel.tsx:64]: >This is that PR :)
🙈

File: src/drafts/SelectPanel2/SelectPanel.tsx (hunk 2 above)
Comment (broccolinisoup) [reply src/drafts/SelectPanel2/SelectPanel.tsx:108]: Ahh that makes sense!!

**Review by broccolinisoup — APPROVED** [review]

**Pull request merged**